src/main/java/com/example/onlinebookstore/model/User.java

```diff
@@ -45,6 +48,10 @@ public class User implements UserDetails {
             inverseJoinColumns = @JoinColumn(name = "role_id"))
     private Set<Role> roles;
 
+    @OneToOne(mappedBy = "user")
+    @Cascade(value = CascadeType.ALL)
+    private ShoppingCart shoppingCart;
+
```

Comment:
redundant, you already has user id in shoppingCart model

Reply:
I think not, because it should be an implementation of logic using cascades. For example, when we remove a user, the shopping cart with its items should automatically be removed as well. The same applies to creation, etc